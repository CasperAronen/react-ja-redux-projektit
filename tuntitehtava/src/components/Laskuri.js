import React from 'react';

const Laskuri = () => {
    const [lkm, asetaLkm] = React.useState(0);
    const [lkm2, asetaLkm2] = React.useState(0);
      return (
        <div>
          <p>Olet lisänny {lkm}</p>
          <p>Olet vähentänyt {lkm2}</p>
          <p>Yhteensä {lkm - lkm2}</p>
          <h1 data-testid="laskuri">{ lkm }</h1>
          <button data-testid="nappi-ylos" onClick={() => asetaLkm(lkm + 1)}>lisää</button>
          <button disabled data-testid="nappi-alas" onClick={() => asetaLkm(0)}>Nollaa</button>    
        </div>
      );
  }
export default Laskuri;