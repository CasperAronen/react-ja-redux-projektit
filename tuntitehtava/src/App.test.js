import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Laskuri from './App'
 
afterEach(cleanup)
 
it('snapshot kokeilu', () => {
    const { asFragment } = render(<Laskuri />);
 
    expect(asFragment(<Laskuri />)).toMatchSnapshot()
});