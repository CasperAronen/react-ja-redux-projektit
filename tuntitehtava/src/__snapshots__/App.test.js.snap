// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot kokeilu`] = `
<DocumentFragment>
  <div>
    <p>
      Olet lisänny 0
    </p>
    <p>
      Olet vähentänyt 0
    </p>
    <p>
      Yhteensä 0
    </p>
    <button>
      lisää
    </button>
    <button>
      vähennä
    </button>
    <button>
      Nollaa
    </button>
    <button>
      Nollaa
    </button>
  </div>
</DocumentFragment>
`;

exports[`snapshot kokeilu 1`] = `
<DocumentFragment>
  <div>
    <p>
      Olet lisänny 0
    </p>
    <p>
      Olet vähentänyt 0
    </p>
    <p>
      Yhteensä 0
    </p>
    <button>
      lisää
    </button>
    <button>
      vähennä
    </button>
    <button>
      Nollaa
    </button>
    <button>
      Nollaa
    </button>
  </div>
</DocumentFragment>
`;
